src/services/swap_service.ts

```diff
@@ -315,6 +326,69 @@ export class SwapService implements ISwapService {
             swapQuoteRequestOpts = ASSET_SWAPPER_MARKET_ORDERS_OPTS;
         }
 
+        const marketSide = sellAmount !== undefined ? MarketOperation.Sell : MarketOperation.Buy;
+        let feeToken = buyToken; // Default fee token is buy token
+
+        // Prepare Sell Token Fees
+        let sellTokenFees: Fees | undefined = undefined; // fees object to return to the caller
+        let sellTokenFeeAmount = ZERO; // total sell token fee to charge
+        let sellTokenFeeAmounts: AffiliateFeeAmount[] = []; // sell token fee amounts used by affiliate fee transformer
+        let sellTokenFeeOnChainTransferGas = ZERO; // the gas cost for transferring sell token fee on-chain
+        if (metaTransactionVersion !== undefined && marketSide === MarketOperation.Sell) {
+            // Use sell token as fee
+
+            // The check is only intended for compile time type checking
```

Comment:
We sometimes just say:
```suggestion
            // Narrow the type
```